File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int goodNodes(TreeNode* root) {
        int pathMax = 0; 
        if(!root)
            return 0;
        stack<pair<TreeNode*, int>> s;

        s.push({root, root->val});
        int count= 0;
        while(!s.empty()){
            TreeNode* n = s.top().first;
            int curMax = s.top().second;
            if(curMax <= n->val){
                count++;
            }
            s.pop();
            if(n->left){
                
                s.push({ n->left, max(n->left->val, curMax) });
            }
            if(n->right){
                
                s.push({ n->right, max(n->right->val, curMax) });
            }
            

        }

        return count;


    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        self.m = 0
        self.recurs(root, 0)
        return self.m
    def recurs(self, root, cur):
        if not root:
            return
        cur +=1
        self.m = max(self.m, cur)
        self.recurs(root.left,cur)
        self.recurs(root.right,cur)